props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,720 @@
+    SaberBase::DoEffect(EFFECT_STUN, 0);
+    shots_fired_++;
+    PVLOG_NORMAL << "******** STUN - Remaining shots = " << GetBulletCount() << "\n";
+  } else if (blaster_mode == MODE_KILL) {
+    SFX_blast.Select(-1);
+    SaberBase::DoEffect(EFFECT_FIRE, 0);
+    shots_fired_++;
+    PVLOG_NORMAL << "******** FIRE - Remaining shots = " << GetBulletCount() << "\n";
+  } else {
+    // Set up the autofire pairing if the font suits it.
+    if (!IsEmpty()) {
+      SelectAutoFirePair();
+      auto_firing_ = true;
+      SaberBase::SetLockup(LOCKUP_AUTOFIRE);
+      SaberBase::DoBeginLockup();
+    }
+  }
+}
+
+void Loop() override {
+  PropBase::Loop();
+  PollNextAction();
+  sound_library_.Poll(wav_player);
+  if (wav_player && !wav_player->isPlaying()) wav_player.Free();
+
+  if (blaster_mode == MODE_AUTO) {
+    // First time through, let bgnauto finish first if we have one
+    if (SFX_bgnauto && GetWavPlayerPlaying(&SFX_bgnauto)) return;
+
+    // No bgnauto or bgnauto has finished playing, so initialize and start auto-firing
+    if (auto_firing_ && !auto_player_) {
+      auto_player_ = GetWavPlayerPlaying(&SFX_auto);
+      shots_fired_++;
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+    // use auto.wav's length to count shots
+    if (auto_firing_ && auto_player_ && (millis() - auto_time_) > auto_sound_length) {
+      auto_sound_length = auto_player_->length() * 1000;
+      shots_fired_++;
+      if (IsEmpty()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_firing_ = false;
+        auto_player_.Free();
+        auto_time_ = millis(); // Reset for the first empty sound
+        return;
+      }
+      auto_time_ = millis();
+      PVLOG_NORMAL << "******** AUTOFIRING - Remaining shots = " << GetBulletCount() << "\n";
+    }
+
+    // This will mimic the hammer running on no bullets at the same rate it was firing.
+    // if there are different length auto.wavs, the last played's length is used.
+    if (!auto_firing_ && IsEmpty() && trigger_is_pressed_ && (millis() - auto_time_) > auto_sound_length) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0);
+      auto_time_ = millis(); // Reset for the next empty sound
+    }
+  }
+}
+
+  void Reload() override {
+    SaberBase::DoEffect(EFFECT_RELOAD, 0);
+    shots_fired_ = 0;
+    is_jammed_ = false;
+  }
+
+  void ClipOut() override {
+    if (max_shots_ != -1) shots_fired_ = max_shots_;
+    SaberBase::DoEffect(EFFECT_CLIP_OUT, 0);
+    no_clip_ = true;
+  }
+
+  void ClipIn() override {
+    SaberBase::DoEffect(EFFECT_CLIP_IN, 0);
+    shots_fired_ = 0;
+    is_jammed_ = false;
+    no_clip_ = false;
+  }
+
+  // Pull in parent's SetPreset, but turn the blaster on.
+  void SetPreset(int preset_num, bool announce) override {
+    PropBase::SetPreset(preset_num, announce);
+      // Just always default to on.
+       if (!SaberBase::IsOn()) On();
```

Comment:
Not sure why always doing this is better than doing it *if* there is an SFX_poweron?

Reply:
I guess I thought it was weird to use the Plecter Monophonic Filename for saber ignition as a basis for whether the blaster has a power on ability? Unless it's supposed to be "advertised" as a supported blaster font sound?
https://pod.hubbe.net/sound/sound-font-configuration.html

Reply:
Ok so the reason this was confusing is because it was explained backwards in the instruction?
According to the actual code, the blaster will not turn ON by default if there is a poweron.wav
```cpp
  // Pull in parent's SetPreset, but turn the blaster on.
  void SetPreset(int preset_num, bool announce) override {
    PropBase::SetPreset(preset_num, announce);

    if (!SFX_poweron) {
      if (!SaberBase::IsOn()) {
        On();
      }
    }
  }
```
Which now makes sense, as in you would want to use your POWER button to control it.
I've incorporated it.